storage_controller/src/node.rs

```diff
@@ -82,6 +88,7 @@ impl Node {
         self.id == register_req.node_id
             && self.listen_http_addr == register_req.listen_http_addr
             && self.listen_http_port == register_req.listen_http_port
+            && self.use_https == register_req.use_https
```

Comment:
What's the path for enabling this? When pageservers start returning `use_https=true`, `registration_match` will return false and we end up [here](https://github.com/neondatabase/neon/blob/main/storage_controller/src/service.rs#L5849-L5860), blocking pageserver start-up. Feels like there should be a code path in `Service::node_register` that detects the `use_https` mismatch and updates the database.